src/utils/saml.ts

```diff
@@ -1194,12 +1202,41 @@ export const preValidateResponse: PreValidateResponseT = (
     )
     .chain(_ =>
       fromEitherToTaskEither(
-        validateIssuer(_.Response, _.SAMLResponseCache.idpIssuer).map(() => _)
-      )
+        validateIssuer(_.Response, _.SAMLResponseCache.idpIssuer).chain(
+          Issuer =>
+            fromOption("Format missing")(
+              fromNullable(Issuer.getAttribute("Format"))
+            ).fold(
+              () => right(_),
+              _1 =>
+                fromPredicate(
+                  FormatValue => !FormatValue || FormatValue === ISSUER_FORMAT,
```

Comment:
this test pass if FormatValue is an empty string (`!"" === true`), I guess that we should compare it with undefined / null

Reply:
Yes, but for some reason, if the Format attribute is missing the method `getAttribute` returns an empty string instead of `null`.
Into the SPID Validator rules the are only two tests for Format attribute of Issuer into the Response element, which are _missing Format_ and _Invalid Format_, so the preValidate works as expected into the tests.